### ISSUE TITLE
static local provider does not (seem to?) resolve http URLs

### ISSUE THREAD

**Issue by mralext20** [Frooxius/NeosPublic#1044]
today i was attempting to mess with new localization features, especially regarding the static locale provider component. 

is it possible to allow remote URLs to work in this component? i'd imagine it's a similar security case to loading images over the network.

**Comment by Frooxius** [comment]
The system is not meant to be used by users at the moment, it will only load localization files from fixed location in the Neos installation. See this weekly update for details: https://store.steampowered.com/newshub/app/740250/view/4582931987118953541

Being able to import custom locales is planned, but this will involve cloud system for Neos to help them be resolved. I want to ensure that the details of the system are settled in before that.

I'm not entirely sure if I want them to be loadable from HTTP, at least not right now. It's problematic on a few fronts:
- I'm not sure I'll keep the json format. With the locales being part of the install, I can easily auto-convert all of them, but if there are going to be external ones, that's going to lock it in for compatibility reasons.
- It requires a convention on how to look them up - e.g. following the naming scheme
- Others cannot contribute locales independently. Only by sending them to whoever can upload them. The Neos system will have ability to dynamically resolve and version the locales.
- People could start building out some hacky solutions on top of it right now, which I want to avoid, because they might be hard to undo without breaking stuff.

**Comment by ProbablePrime** [comment]
I agree on all of these points except for: "Others cannot contribute locales independently. Only by sending them to whoever can upload them. The Neos system will have ability to dynamically resolve and version the locales."

If im working on a project for Neos, I might need an entirely new locale set which is very specific to my project. My project members should be able to submit localization files and updates but not EVERY user needs to be able to. In some Business to Business scenarios involving projects with clients and customers it may also be desirable to limit the local management to only the company working on the project etc.

Supplying my own locale files via HTTP or any other method which are separate from the main locale set is more suitable here for this kind of situation.

For strings used within Neos' Ui and creations I agree though, its just for those custom scenarios i mention that HTTP or some other method of segregated UI files makes sense.

**Comment by Frooxius** [comment]
It's a good point, I definitely don't want the system to be enforced where it would be detrimental or inconvenient. My thought is that by default any 3rd party translations are considered "unofficial" and will be offered as such. As the original owner of the main translation, you'd be able to specify some of them as official and have it automatically loaded in all cases if it matches the user's locale.

That way people can't just sneak in translations for your content, while looking official. If you wanted to have full control, you'd only mark the official ones as the ones you uploaded. 

Question is, would it be needed to completely disable ability for anyone to contribute a translation for your content, even if it's marked as unofficial and requires user action to be loaded?

**Comment by ProbablePrime** [comment]
For your last question, in most cases no but in some yes.

For businesses there are most likely legal(im not a lawyer) and accessibility(I'm not an expert here) regulations  about the translations of your project which mean that they have to be done by a professional company or service.

Microsoft has its own processes here that are very complicated but get very good translations for all languages. 

Larger companies are probably not something you'll have to worry about for a while so for now the plan you mention sounds good for the main Neos user base and experience. I just wanted to surface the larger issues that might come up so that you were aware.

Thank you!

**Comment by Frooxius** [comment]
Ah that makes sense, thanks! I think in that case it could be just configurable to disallow 3rd party translations completely for the Neos system. Having some way to store them on your own server would be possible too, but I think I'd add it afterwards. 

It might be added now and be relatively quick, but without the proper system I'm not sure what kind of impact it'll have, I want people to use the proper Neos-managed system wherever possible, since that'll ensure long term compatibility for the content and most flexibility.

**Comment by ProbablePrime** [comment]
I don't require it now but i was just thinking while using the MMC strings. :D
